webui/react/src/hooks/useKeyTracker.ts

```diff
@@ -0,0 +1,39 @@
+import { EventEmitter } from 'events';
+
+import { useCallback, useEffect } from 'react';
+
+export enum KeyEvent {
+  KeyUp = 'KeyUp'
+}
+
+type KeyListener = (e: KeyboardEvent) => void;
+
+export const keyEmitter = new EventEmitter();
+
+const externalListeners: KeyListener[] = [];
+
+const useKeyTracker = (): void => {
+  const handleKeyUp = useCallback((e: KeyboardEvent) => {
+    if (e.target) {
+      const element = e.target as Element;
+      if ([ 'input', 'textarea' ].includes(element.tagName.toLowerCase())) return;
+      if (element.getAttribute('contenteditable')) return;
+    }
+    keyEmitter.emit(KeyEvent.KeyUp, e);
+  }, []);
+
+  useEffect(() => {
+    if (externalListeners.length !== 0) return;
```

Comment:
This is to make sure that if `useKeyTracker()` is called more than 1 time in the whole app, only 1 key tracker event listener/broadcaster runs, right?

Reply:
yeah exactly, I was trying to think of a singleton pattern to use for hooks but it seems like that is considered an anti-pattern in React. I did a slight update to just keep track of count because I am not actually using the array of listeners for removal as I originally intended, just to keep the tracker size smaller.